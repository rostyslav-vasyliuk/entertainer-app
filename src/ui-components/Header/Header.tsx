import React from 'react';
import { Header, Body, Left, Right, Title } from 'native-base';
import { Text, TouchableOpacity } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import { HEADER_BACKGROUND, TEXT_COLOR, TEXT_COLOR_SECONDARY } from '../../constants/color-constants';

const HeaderCustom = (props) => {
  return (
    <Header iosBarStyle='light-content' transparent style={{ backgroundColor: HEADER_BACKGROUND }}>
      <Left>
        {props.back && (
          <TouchableOpacity onPress={props.back}>
            <MaterialIcons name='arrow-back' size={24} color={TEXT_COLOR} style={{ paddingLeft: 4 }} />
          </TouchableOpacity>
        )
        }
      </Left>
      <Body>
        <Title style={{ color: TEXT_COLOR, fontSize: 16, letterSpacing: 1.1, fontWeight: '600', overflow: 'visible', minWidth: 250 }}>
          {props.label}
        </Title>
      </Body>
      <Right>

      </Right>
    </Header>
  )
}

export default HeaderCustom;
