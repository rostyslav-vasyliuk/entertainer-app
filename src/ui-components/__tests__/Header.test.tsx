import React from 'react';
import Header from '../Header/Header';
import renderer from 'react-test-renderer';

describe('<Header />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it('has 1 child', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree.children.length).toBeGreaterThanOrEqual(1);
  });

  it('rendered', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree.children).toBeTruthy();
  });

  it('to match snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
