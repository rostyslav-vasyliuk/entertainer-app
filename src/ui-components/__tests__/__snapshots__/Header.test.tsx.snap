// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> to match snapshot 1`] = `
<View
  onLayout={[Function]}
>
  <View
    iosBarStyle="light-content"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "transparent",
          "borderBottomWidth": 0.5,
          "elevation": 0,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": 0,
        },
        Object {
          "backgroundColor": "#17191c",
        },
      ]
    }
    transparent={true}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
              "letterSpacing": 1.1,
              "minWidth": 250,
              "overflow": "visible",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    />
  </View>
</View>
`;
